modules/uid2IdSystem_shared.js

```diff
@@ -200,11 +613,31 @@ export function Uid2GetId(config, prebidStorageManager, _logInfo, _logWarn) {
       storedTokens = null;
     }
   }
-  // At this point, any legacy values or superseded stored tokens have been nulled out.
+
+  if (cstgIdentity) {
+    if (storedTokens) {
+      const identityKey = Object.keys(cstgIdentity)[0]
+      if (!storedTokens.originalIdentity || storedTokens.originalIdentity[identityKey] !== cstgIdentity[identityKey]) {
+        _logInfo('CSTG supplied new identity - ignoring stored value.', storedTokens.originalIdentity, cstgIdentity);
+        // Stored token wasn't originally sourced from the provided identity - ignore the stored value. A new user has logged in?
+        storedTokens = null;
+      }
+    }
+    if (!storedTokens || Date.now() > storedTokens.latestToken.refresh_expires) {
+      const promise = generateTokenAndStore(config.apiBaseUrl, config.cstg, cstgIdentity, config.clientId, storageManager, _logInfo, _logWarn);
```

Comment:
This looks like we're always calling the CSTG API, even if we have a refresh token?

Reply:
We only call CSTG API if there is no stored token or the stored token cannot be refreshed (refresh_expires < Date.now)